spec/rubocop/ast/processed_source_spec.rb

```diff
@@ -464,4 +464,98 @@ def some_method
       expect(processed_source.following_line(brace_token)).to eq '# line 3'
     end
   end
+
+  describe '#node_tokens' do
+    context 'when no heredoc present' do
+      let(:source) { <<~RUBY }
+        foo(1, 2)
+        bar(3)
+      RUBY
+
+      it 'returns tokens for node' do
+        node = ast.children[1]
+        tokens = processed_source.node_tokens(node)
+
+        expect(tokens.map(&:text)).to eq(['bar', '(', '3', ')'])
+      end
+    end
+
+    context 'when heredoc as argument is present' do
+      let(:source) { <<~RUBY }
+        foo(1, [before], <<~DOC, [after])
+          inside heredoc.
+        DOC
+        bar(2)
+      RUBY
+
+      it 'returns tokens for node before heredoc' do
+        node = ast.children[0].arguments[1]
+        tokens = processed_source.node_tokens(node)
+
+        expect(tokens.map(&:text)).to eq(['[', 'before', ']'])
+      end
+
+      it 'returns tokens for heredoc node' do
+        node = ast.children[0].arguments[2]
```

Comment:
Aren't the heredoc's tokens a bigger issue for `#{}` within the heredoc?

Reply:
Didn't get the point. Can you reword/explain more?

Reply:
I was wondering: would your test fail if the tokens were not sorted? I thought it was for cases where there's a heredoc with for `#{}` that doing the `bsearch` on unsorted tokens might fail.

Reply:
When tokens are not sorted, for some cases, depending on the source, it will fail, sometimes not. It depends on how lucky bsearch will be to find correct locations.
 When they are sorted, it will always succeed.